protecode/util.py

```diff
@@ -310,3 +315,18 @@ def filter_func(other_dict: typing.Dict[str, str]):
         return result.product_id()
     else:
         return None
+
+
+def image_ref_with_digest_if_possible(
+    image_reference: str | oci.model.OciImageReference,
```

Comment:
for OCI Images, it is always possible to calculate digest (the digest is calculated as the SHA-256-digest from the image's manifest (`f'sha256:{hashlib.sha256(oci_client.manifest_raw().content).hexdigest()}'`). How about doing so? We can still omit doing so in cases where we e.g. do not have an OCI Client (however, if we do not have such a client, we will not be able to upload the OCI Image, so I assume this case is actually not relevant in practise).